RASA2.py

```diff
@@ -0,0 +1,465 @@
+                'MOLECULAR_WEIGHTS': 75},
+        'His': {'TO_1': 'E',
+                'PROTEIN_TO_RNA_COMBINATION': {'CAU', 'CAC'},
+                'PKA_AMINOACIDS': [1.82, 9.17],
+                'MOLECULAR_WEIGHTS': 155},
+        'Ile': {'TO_1': 'I',
+                'PROTEIN_TO_RNA_COMBINATION': {'AUU', 'AUC', 'AUA'},
+                'PKA_AMINOACIDS': [2.36, 9.68],
+                'MOLECULAR_WEIGHTS': 131},
+        'Leu': {'TO_1': 'L',
+                'PROTEIN_TO_RNA_COMBINATION': {'CUU', 'CUC', 'CUA', 'CUG'},
+                'PKA_AMINOACIDS': [2.36, 9.60],
+                'MOLECULAR_WEIGHTS': 131},
+        'Lys': {'TO_1': 'K',
+                'PROTEIN_TO_RNA_COMBINATION': {'AAA', 'AAG'},
+                'PKA_AMINOACIDS': [2.18, 8.95, 10.53],
+                'MOLECULAR_WEIGHTS': 146},
+        'Met': {'TO_1': 'M',
+                'PROTEIN_TO_RNA_COMBINATION': {'AUG'},
+                'PKA_AMINOACIDS': [2.28, 9.21],
+                'MOLECULAR_WEIGHTS': 149},
+        'Phe': {'TO_1': 'F',
+                'PROTEIN_TO_RNA_COMBINATION': {'UUU', 'UUC'},
+                'PKA_AMINOACIDS': [2.20, 9.13],
+                'MOLECULAR_WEIGHTS': 165},
+        'Pro': {'TO_1': 'P',
+                'PROTEIN_TO_RNA_COMBINATION': {'CCU', 'CCC', 'CCA', 'CCG'},
+                'PKA_AMINOACIDS': [1.99, 10.96],
+                'MOLECULAR_WEIGHTS': 115},
+        'Ser': {'TO_1': 'S',
+                'PROTEIN_TO_RNA_COMBINATION': {'UCU', 'UCC', 'UCA', 'UCG'},
+                'PKA_AMINOACIDS': [2.21, 9.15],
+                'MOLECULAR_WEIGHTS': 105},
+        'Thr': {'TO_1': 'T',
+                'PROTEIN_TO_RNA_COMBINATION': {'ACU', 'ACC', 'ACA', 'ACG'},
+                'PKA_AMINOACIDS': [2.11, 9.62],
+                'MOLECULAR_WEIGHTS': 119},
+        'Tyr': {'TO_1': 'W',
+                'PROTEIN_TO_RNA_COMBINATION': {'UAU', 'UAC'},
+                'PKA_AMINOACIDS': [2.20, 9.11, 10.07],
+                'MOLECULAR_WEIGHTS': 181},
+        'Trp': {'TO_1': 'Y',
+                'PROTEIN_TO_RNA_COMBINATION': {'UGG'},
+                'PKA_AMINOACIDS': [2.38, 9.39],
+                'MOLECULAR_WEIGHTS': 204},
+        'Val': {'TO_1': 'V',
+                'PROTEIN_TO_RNA_COMBINATION': {'GUU', 'GUC', 'GUA', 'GUG'},
+                'PKA_AMINOACIDS': [2.32, 9.62],
+                'MOLECULAR_WEIGHTS': 117},
+    }
+
+    TRANSCRIBE_DICT: dict = {'A': 'A',
+                             'U': 'T',
+                             'G': 'G',
+                             'C': 'C',
+                             'a': 'a',
+                             'u': 't',
+                             'g': 'g',
+                             'c': 'c'}
+
+    TO_3_DICT = {nested_dict['TO_1']: key for key,
+    nested_dict in AMINOACIDS_DICT.items()}
+
+    def is_valid_alphabet(self) -> bool:
+        for letter in self.sequence:
+            if letter not in self.TRANSCRIBE_DICT.keys():
+                return False
+        return True
```

Comment:
Монжо чуть чуть изменить эту функцию, так она будет работать побыстрее
```suggestion
    valid_alphabet_set = set(self.TRANSCRIBE_DICT.keys())
    return set(self.sequence).issubset(valid_alphabet_set)
```

Reply:
Отличное замечание. Даже тут не только быстрее, но и просто более питонистый стиль. Не вручную копаем отдельные элементы коллекции, а используем более высокоуровневые методы самих коллекций.